server/py/services/api/launcher.py

```diff
@@ -240,17 +244,25 @@ def _pre_run_node_selector_enrichement(self, runtime, run):
             run.spec.node_selector = resolved_node_selectors
         return run
 
-    def _pre_run_image_pull_secret_enrichment(self, run):
+    def _pre_run_image_pull_secret_enrichment(
+        self,
+        runtime: "mlrun.runtimes.base.BaseRuntime",
+        run: mlrun.run.RunObject,
+    ):
         """
-        Enrich the run object with the project's image pull secret.
+        Enrich both the run object and runtime with the project's image pull secret.
         This ensures the image pull secret is correctly set on the run,
         either from the run spec or from the MLRun config
         """
         existing_image_pull_secret = getattr(run.spec, "image_pull_secret", None)
+        if existing_image_pull_secret is not None:
+            existing_image_pull_secret = existing_image_pull_secret.default
         run.spec.image_pull_secret = (
             existing_image_pull_secret
-            or mlrun.config.config.function.spec.image_pull_secret
+            or mlrun.config.config.function.spec.image_pull_secret.default
         )
+
+        runtime.spec.image_pull_secret = run.spec.image_pull_secret
```

Comment:
if the run is enriched with the image pull secret, do you need it on the runtime as well?